pkg/api/types.go

```diff
@@ -307,6 +307,7 @@ type TestStepConfiguration struct {
 	OpenshiftInstallerCustomTestImageClusterTestConfiguration *OpenshiftInstallerCustomTestImageClusterTestConfiguration `json:"openshift_installer_custom_test_image,omitempty"`
 }
 
+// TestStep is an internal representation of a full test step that ci-operator can run
 type TestStep struct {
 	Name        string               `json:"name,omitempty"`
```

Comment:
After moving this, the internal types also won't need `json` annotations as they are not read from config